packages/react-native/ReactAndroid/src/main/java/com/facebook/react/devsupport/HMRClient.java

```diff
@@ -29,6 +29,18 @@ public interface HMRClient extends JavaScriptModule {
    */
   void setup(String platform, String bundleEntry, String host, int port, boolean isEnabled);
 
+  /**
+   * Enable the HMRClient so that the client will receive updates from Metro.
+   *
+   * @param platform The platform in which HMR updates will be enabled. Should be "android".
+   * @param bundleEntry The path to the bundle entry file (e.g. index.ios.bundle).
+   * @param host The host that the HMRClient should communicate with.
+   * @param port The port that the HMRClient should communicate with on the host.
+   * @param isEnabled Whether HMR is enabled initially.
+   * @param scheme The protocol that the HMRClient should communicate with on the host (defaults to http).
+   */
```

Comment:
Heads up that this is a breaking change for any app/library that is implementing HMRClient is OSS.

Reply:
Is there a way to make this not a breaking change? The React Native HMR client "spec" allows both - while Android only allowed 1.

Not sure how we can do that in Java, making something optional

Reply:
Wait, are you sure that this is a breaking change? Even if HMR Clients do not use the `scheme` prop, it still shouldn't be an issue in the JS side of the HMR client.

Reply:
Yup, as @javache pointed out, you're not supposed to implement this interface directly. I found no meaningful implementation of it, so it's safe to assume this is not a breaking change.